libraries/abstractions/transport/secure_sockets/transport_secure_sockets.h

```diff
@@ -0,0 +1,188 @@
+    TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER,   /**< At least one parameter was invalid. */
+    TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY, /**< Insufficient memory required to establish connection. */
+    TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, /**< Provided credentials were invalid. */
+    TRANSPORT_SOCKET_STATUS_API_ERROR,           /**< A call to a system API resulted in an internal error. */
+    TRANSPORT_SOCKET_STATUS_DNS_FAILURE,         /**< Resolving hostname of the server failed. */
+    TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE      /**< Initial connection to the server failed. */
+} TransportSocketStatus_t;
+
+
+/**
+ * @brief Information on the remote server for connection setup.
+ */
+typedef struct ServerInfo
+{
+    const char * pHostName; /**< @brief Server host name. */
+    size_t hostNameLength;  /**< @brief Length of the server host name. */
+    uint16_t port;          /**< @brief Server port in host-order. */
+} ServerInfo_t;
+
+
+/**
+ * @brief Contains the credentials necessary for connection setup.
+ *
+ */
+typedef struct SocketsConfig
+{
+    bool enableTls;         /**< @brief Whether require TLS for the transport. */
+    uint32_t sendTimeoutMs; /**< @brief Timeout for transport send. */
+    uint32_t recvTimeoutMs; /**< @brief Timeout for transport recv. */
+
+    /**
+     * @brief Set this to a non-NULL value to use ALPN.
+     *
+     * This string must be NULL-terminated.
+     *
+     * See [this link]
+     * (https://aws.amazon.com/blogs/iot/mqtt-with-tls-client-authentication-on-port-443-why-it-is-useful-and-how-it-works/)
+     * for more information.
+     */
+    const char * pAlpnProtos;
+
+    /**
+     * @brief Length of the ALPN protocols array.
+     */
+    uint32_t alpnProtosLen;
+
+    /**
+     * @brief Disable server name indication (SNI) for a TLS session.
+     */
+    bool disableSni;
+
+    /**
+     * @brief Set this to a non-zero value to use TLS max fragment length
+     * negotiation (TLS MFLN).
+     *
+     * @note The network stack may have a minimum value for this parameter and
+     * may return an error if this parameter is too small.
+     */
+    size_t maxFragmentLength;
+
+    const char * pRootCa;     /**< @brief String representing a trusted server root certificate. */
+    size_t rootCaSize;        /**< @brief Size associated with #IotNetworkCredentials_t.pRootCa. */
+    const char * pClientCert; /**< @brief String representing the client certificate. */
+    size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials_t.pClientCert. */
+    const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
+    size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials_t.pPrivateKey. */
+} SocketsConfig_t;
+
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
```

Comment:
If this supports supports a TCP only connection as well (i.e. without TLS), then the comment can be updated accordingly

Reply:
fixed iin 9e9a7389c3e1f06de838bfe2cd3e5254c1cc8bb1